dev/cmd/dev.py

```diff
@@ -1,40 +1,39 @@
-import argparse
-import os
-import sys
-
+import glob
+import os.path
 import spack
 import spack.cmd
-import llnl.util.tty as tty
 
-SPACKDEV_FILE = os.path.realpath(os.path.expanduser(__file__))
-SPACKDEV_LIB = os.path.realpath(os.path.join(os.path.dirname(SPACKDEV_FILE),
-                                             '..', '..', 'lib'))
-sys.path.insert(0, SPACKDEV_LIB)
+# Import sub-commands
+subcommands = []
+
+
+def import_subcommands_automatically():
+    subcommand_glob = os.path.join(os.path.dirname(__file__), '..', 'subcommands', '*.py')
+    command_files = glob.glob(subcommand_glob)
+    subcommands.extend(filter(
+        lambda x: not x.startswith('__'), [os.path.basename(x)[:-3] for x in command_files]
+    ))
+    t = __import__('subcommands', globals(), locals(), subcommands, 2)
+    for name in subcommands:
+        globals()[name] = getattr(t, name)
+
+import_subcommands_automatically()
```

Comment:
> I spoke with @chissg today and I think the thing I missed was that he didn't want to have to refer to the commands explicitly: the idea is that he could add more subcommands and those could be imported here without updating this code.

@chissg The change above permits to do what you ask for without changes to Spack `develop`. If you implement another subcommand in the `subcommands` folder then  `dev.py` does not need any change to import it and make it available. I personally prefer the previous version with the explicit import, but this is just to demonstrate that adding this capability to an extension doesn't need any change to current Spack.